### ISSUE TITLE
Design Guidelines: HTTP Status Code type should be Integer, not String

### ISSUE THREAD

**Issue by hdamker** [camaraproject/WorkingGroups#143]
Currently the [API Design Guidelines](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md) define HTTP Status Code as String within the "proposed json error structure":

```
{
   "status": "400",
   "code": "INVALID_ARGUMENT",
   "message": "A human readable description of what the event represent"
}
```

There is a discussion in QualityOnDemand (see [here](https://github.com/camaraproject/QualityOnDemand/pull/104#discussion_r1092017589)) which came to the conclusion that the type Integer for the HTTP Status Code would be the better solution:
-  Easier to use for API consumers (as all HTTP Status Codes are three digit integer between 100 and 599)
-  Compliant to RFC 9110 which says: [The status code of a response is a three-digit integer code](https://www.rfc-editor.org/rfc/rfc9110.html#name-status-codes:~:text=The%20status%20code%20of%20a%20response%20is%20a%20three%2Ddigit%20integer%20code)

It would be great if this issue could be addressed short-term in Commonatlities as it is currently blocking the PR [Align error format with Commonalities guidelines](https://github.com/camaraproject/QualityOnDemand/pull/104#top) which itself should be done before other PRs.

---

Further noticed, but not part of this issue:
- The references to RFC 2616 and RFC 7231 in [chapter 3.2](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md#32-http-response-codes) of the API Design Guidelines could be replaced by [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110), as they are obsoleted by the newer one.
- Chapters 3.2 (HTTP Response Codes), 6 (Error Responses) and 11.5 (Data Definitions) have overlapping content and are not consistent

**Comment by shilpa-padgaonkar** [comment]
I agree. As stated in https://httpwg.org/specs/rfc9110.html#rfc.section.15 - status code of a response should be a three-digit integer code.
I would recommend the change as

{
   "code": 400,
   "status": "INVALID_ARGUMENT",
   "message": "A human readable description of what the event represent"
}

If the colleagues currently involved in the discussion for aligning QoD with the Guidelines doc, already create a PR against the doc, it would help speed up this task as it can then be discussed and agreed in the next commonalities call.

**Comment by eric-murray** [comment]
Agreed

**Comment by RubenBG7** [comment]
Agree, let me change the definition

**Comment by RubenBG7** [comment]
Issues 145 (API Guidelines) and 146 (CAMARA_common.json ModelError Property) with docs changes applied.

**Comment by hdamker** [comment]
Issue is resolved with #145 and #146

**Comment by RubenBG7** [comment]
Changes added on #149 to delete string patter due to change to integer, and #150 fixed some bad words and format in github view
